openedx/core/djangoapps/profile_images/images.py

```diff
@@ -0,0 +1,128 @@
+"""
+Image file manipulation functions related to profile images.
+"""
+from cStringIO import StringIO
+
+from django.conf import settings
+from django.core.files.base import ContentFile
+from PIL import Image
+
+from ..user_api.accounts.api import get_profile_image_storage
+
+
+class DevMsg(object):
+    """
+    Holder for pseudo-constants.
+    """
+    FILE_TOO_LARGE = 'Maximum file size exceeded.'
+    FILE_TOO_SMALL = 'Minimum file size not met.'
+    FILE_BAD_TYPE = 'Unsupported file type.'
+    FILE_BAD_EXT = 'File extension does not match data.'
+    FILE_BAD_MIMETYPE = 'Content-Type header does not match data.'
+
+
+class ImageFileRejected(Exception):
```

Comment:
Good call, I will rename to ImageValidationError.  I do find it advisable to maintain the custom exception since it helps us make a very clear distinction between exit conditions we have explicitly coded for and unexpected ones, which should percolate up to become server errors.